Content/Items/Dungeon/Weapons.ThousandthDegree.cs

```diff
@@ -0,0 +1,698 @@
+                if (proj.ModProjectile is ThousandthDegreeProjectileFired wheel)
+                {
+                    wheel.parentProj = Projectile;
+                    wheel.inputHeat = CurrentHeat;
+                }
+                fireProjectile = proj;
+            }
+
+            for (int i = 0; i < 8; i++)
+            {
+                Vector2 velo = Projectile.DirectionTo(Main.MouseWorld);
+                Dust.NewDustPerfect(wheelPos + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.MagmaSmoke>(), velo.RotatedByRandom(0.5f) * Main.rand.NextFloat(1.8f, 2.5f) + (Vector2.UnitY * -1f), 155, default, Main.rand.NextFloat(0.6f, 0.9f));
+                
+                Dust.NewDustPerfect(wheelPos, ModContent.DustType<Dusts.Glow>(), velo.RotatedByRandom(0.65f) * Main.rand.NextFloat(5f, 7f), 0, new Color(255, 160, 50), 0.4f);
+                
+                Dust.NewDustPerfect(wheelPos, ModContent.DustType<Dusts.Glow>(), velo.RotatedByRandom(0.65f) * Main.rand.NextFloat(6f, 8f), 0, new Color(255, 50, 15), 0.45f);
+                
+                Dust.NewDustPerfect(wheelPos, DustID.Torch, velo.RotatedByRandom(0.65f) * Main.rand.NextFloat(6f, 8f), 0, default, 1.35f);
+            }
+
+            Helper.PlayPitched("Magic/FireHit", 0.45f, 0, wheelPos);
+            Core.Systems.CameraSystem.Shake += 4;
+        }
+    }
+
+    class ThousandthDegreeProjectileFired : ModProjectile
+    {
+        private const int MAXHEAT = 30;
+
+        private List<Vector2> cache;
+        private Trail trail;
+        private Trail trail2;
+
+        public Projectile parentProj;
+
+        public float inputHeat;
+        public int lerpTimer;
+        public int switchDelay;
+
+        public bool returning;
+        public bool switched;
+
+        public Vector2 moveDirection;
+        public Vector2 newVelocity = Vector2.Zero;
+        public float speed => MathHelper.Lerp(3f, 10f, speedTimer / 65f);
+        public int speedTimer;
+
+        bool collideX = false;
+        bool collideY = false;
+        bool collided;
+
+        public Player owner => Main.player[Projectile.owner];
+        public override string Texture => AssetDirectory.DungeonItem + Name;
+
+        public override void SetStaticDefaults()
+        {
+            DisplayName.SetDefault("Blazing Wheel");
+        }
+
+        public override void SetDefaults()
+        {
+            Projectile.DamageType = DamageClass.Melee;
+            Projectile.friendly = true;
+            Projectile.penetrate = -1;
+
+            Projectile.tileCollide = true;
+
+            Projectile.width = Projectile.height = 36;
+            Projectile.timeLeft = 2400;
+            Projectile.ignoreWater = true;
+        }
+        public override void AI()
```

Comment:
This logic seems fairly complicated -- it would probably be wise to attempt to break out behavior into seperate functions or to atleast document these large conditional blocks

Reply:
I would like to see some encapsulation for the seperate 'phases' of the wheels lifetime here. Ideally each has their behavior in a method and then the conditions for that method to be called are checked in AI. This logic is really difficult to digest whole.